src/Controller/CategoryItemController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Controller;
+
+use App\Model\CategoryItemManager;
+
+class CategoryItemController extends AbstractController
+{
+    public function index(): string|null
+    {
+        $categoryItemManager = new CategoryItemManager();
+        $categoriesItems = $categoryItemManager->selectAll('id');
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            if ($_POST['type'] === 'add') {
+                $_POST['title'] =  ucfirst($_POST['title']);
+                $_POST['description'] = ucfirst($_POST['description']);
+                // clean $_POST data
+                $categoryItem = array_map('trim', $_POST);
+
+                // TODO validations (length, format...)
+                $errors = $this->checkdata($categoryItem);
+
+
+                // if validation is ok, insert and redirection
+                if (!empty($errors)) {
+                    return $this->twig->render('CategoryItem/index.html.twig', [
+                        "errors" => $errors,
+                        "categoriesItems" => $categoriesItems
+                    ]);
+                } else {
+                    $categoryItemManager = new CategoryItemManager();
+                    $categoryItemManager->insert($categoryItem);
+                    header('Location:/categories_items');
+                    return null;
+                }
+            }
+            if ($_POST['type'] === 'select') {
+                $errors2 = [];
+
+                // Retrieve selected categories
+                $categoriesSelect = $this->insertPostCategory();
+
+                // Checked errors
+                if (count($categoriesSelect) < 4 || count($categoriesSelect) > 4) {
+                    $errors2[] = "Tu as sélectionné " .
+                        count($categoriesSelect) .
+                        " catégorie(s) tu dois sélectionner 4 catégories";
+                }
+
+                // return $errors2;
+                if (!empty($errors2)) {
+                    return $this->twig->render('CategoryItem/index.html.twig', [
+                        "errors2" => $errors2,
+                        "categoriesItems" => $categoriesItems
+                    ]);
+                } else {
+                    $categoryItemManager = new CategoryItemManager();
+                    $categoryItemManager->updateNotIncarousel();
+                    $categoryItemManager->updateIncarousel($categoriesSelect);
+                    header('Location:/');
+                }
+            }
+        }
+        return $this->twig->render('CategoryItem/index.html.twig', ['categoriesItems' => $categoriesItems]);
+    }
+
+    public function checkdata($data): array
```

Comment:
if this function does not need to be used outside the class, change it as a private method